src/model/internal-messages.model.ts

```diff
@@ -10,7 +10,8 @@ export interface LoginData {
 export interface RegisterData {
   username: string
   password: string
-  privateKey: PrivateKey
+  privateKey?: PrivateKey
+  mnemonic?: Mnemonic
```

Comment:
I would make two different `RegisterData` interface like `RegisterWithPrivateKey` and `RegisterWithMnemonic` where the properties are required. Thereby, you can create `type RegisterData = RegisterWithPrivateKey | RegisterWithMnemonic`. if you let it as it is, you allow to privateKey and mnemonic be null by type definition that will throw error